overlord/devicestate/devicemgr.go

```diff
@@ -797,15 +797,52 @@ func systemFromSeed(label string, current *seededSystem) (*System, error) {
 		Brand:   brand,
 		Actions: defaultSystemActions,
 	}
-	if current != nil && isCurrentSystem(current, &system) {
```

Comment:
not sure I understand this change?  also if we did this we don't need to pass current anymore but then it seems we get code duplication

Reply:
Ahh, it's part of the next piece that isn't published yet. I'll tweak it here and rebase the other one.

Reply:
Dropped `current` form parameters now.